src/chat/participant.ts

```diff
@@ -161,21 +142,92 @@ function filterContextHistory(
   if (filteredHistory.length === 0) {
     return [];
   }
-
   const messages: LanguageModelChatMessage[] = [];
   for (const turn of filteredHistory) {
     // don't re-use previous prompts since the model may misinterpret them as part of the current prompt
     if (turn instanceof ChatRequestTurn) {
-      // TODO: check for references/commands used?
+      if (turn.participant === PARTICIPANT_ID) {
+        messages.push(LanguageModelChatMessage.User(turn.prompt));
+      }
       continue;
     }
     if (turn instanceof ChatResponseTurn) {
       // responses from the participant:
       if (turn.response instanceof ChatResponseMarkdownPart) {
-        messages.push(LanguageModelChatMessage.User(turn.response.value.value, PARTICIPANT_ID));
+        messages.push(
+          LanguageModelChatMessage.Assistant(turn.response.value.value, PARTICIPANT_ID),
+        );
       }
     }
   }
 
   return messages;
 }
+
+export async function handleChatMessage(
```

Comment:
This would be where we would inject dynamic context that we discussed today @shouples ?

Reply:
Yep, I think between here and also up at the beginning of the chat request flow in https://github.com/confluentinc/vscode/blob/bdb5e72bd838c6c017e1174913cb7e235e4296b0/src/chat/participant.ts#L40

May depend on exactly what context we need to add though, versus which context we should allow the model to pull (via tool call) itself.